File: pset6/vigenere.py
import cs50
import sys

def main():
    if len(sys.argv) != 2:
        print("Usage: ./vigenere k")
        exit(1)
    
    if sys.argv[1].isalpha() == False:
        print("Usage: ./vigenere k")
        exit(1)
    
    print("plaintext: ",end='')   
    message = cs50.get_string()
    translated = []
    keyIdx = 0
    keylen = len(sys.argv[1])
    
    for symbol in message:
        if symbol.isalpha():
            key = ord(sys.argv[1][keyIdx % keylen].lower()) - 97
            keyIdx += 1
            translated.append(caesar(symbol, key))
        else:
            translated.append(symbol)
    print("ciphertext: ",end='');
    print("".join(translated))
    exit(0)
    
def caesar(char, key):
    if char.isupper():
        return chr(((ord(char) - 65 + key) % 26) + 65)
    else:
        return chr(((ord(char) - 97 + key) % 26) + 97)

if __name__ == "__main__":
    main()

File: pset2/vigenere/vigenere.c
#include <stdio.h>
#include <ctype.h>
#include <cs50.h>
#include <string.h>

int main(int argc,string argv[]){
    if(argc!=2){
        printf("Error:");
        return 1;
    }
    string k = argv[1];
    for(int i = 0; i< strlen(k); i++){
        if(!isalpha(k[i])){
            printf("Error:");
            return 1;
        }
    }
    printf("plaintext: ");
    string p = get_string();
    int size = strlen(p);
    char c[size];
    int j = 0;
    int key;
    for(int i = 0; i<size; i++){
        j = j%strlen(k);
        if(isalpha(p[i])){
            if(isupper(k[j])){
                key = k[j]-'A';
            }
            else{
                key = k[j] - 'a';
             }
            if(isupper(p[i])){
                c[i] = (p[i]-'A'+key)%26 + 'A';
            }
            else{
                c[i] = (p[i]-'a'+key)%26 + 'a';
            }
            j++;
        }
        else{
            c[i] = p[i];
        }
    }
    printf("ciphertext: ");
    for(int i = 0; i<size; i++){
        printf("%c",c[i]);
    }
    printf("\n");
    return 0;
}